### ISSUE TITLE
[Flake] vcpkg Failed to take the filesystem lock

### ISSUE THREAD

**Issue by devjgm** [googleapis/google-cloud-cpp#4487]
In the ~Windows~ Docker quickstart cmake presubmit build:

https://source.cloud.google.com/results/invocations/152e696f-0409-447d-bfe7-19c052095160/targets/cloud-cpp%2Fgithub%2Fgoogle-cloud-cpp%2Fmaster%2Fdocker%2Fquickstart-cmake-presubmit/log

```
[76/77] Building CXX object CMakeFiles/vcpkglib.dir/src/vcpkg/vcpkglib.cpp.o
[77/77] Linking CXX executable vcpkg
Telemetry
---------
vcpkg collects usage data in order to help us improve your experience.
The data collected by Microsoft is anonymous.
You can opt-out of telemetry by re-running the bootstrap-vcpkg script with -disableMetrics,
passing --disable-metrics to vcpkg on the command line,
or by setting the VCPKG_DISABLE_METRICS environment variable.

Read more about vcpkg telemetry at docs/about/privacy.md
There are no outdated packages.
Waiting to take filesystem lock on /h/vcpkg-quickstart/.vcpkg-root...
Failed to take the filesystem lock on /h/vcpkg-quickstart/.vcpkg-root:
    Device or resource busy
Exiting now.
2020-07-08T21:44:00Z: Docker run failed with exit status 1.
```

**Comment by devjgm** [comment]
Seems to be related to https://github.com/microsoft/vcpkg/issues/12286 (filed 2 days ago) and may be fixed in https://github.com/microsoft/vcpkg/issues/12286

**Comment by coryan** [comment]
https://source.cloud.google.com/results/invocations/7fcaf995-e3f7-4bc7-8fc8-107974609db5

**Comment by ras0219-msft** [comment]
Cross-posted to https://github.com/microsoft/vcpkg/issues/12286:

> docker .... --volume /tmpfs/src/github/google-cloud-cpp/cmake-out/home/ci-ubuntu-18.04-quickstart-cmake:/h ....
...
Waiting to take filesystem lock on /h/vcpkg-quickstart/.vcpkg-root...
Failed to take the filesystem lock on /h/vcpkg-quickstart/.vcpkg-root:
Device or resource busy
Exiting now.

It looks like the file locking APIs we use are unavailable in a mapped docker volume. This looks like a straightforward case of "we should offer an environment variable to not take the lock". Users in docker could then pass `--env VCPKG_SKIP_LOCKING=1`.

**Comment by strega-nil** [comment]
This should be fixed in microsoft/vcpkg#12227.

**Comment by devjgm** [comment]
As pointed about above, this is fixed by microsoft/vcpkg#12227. Thanks to @strega-nil for the fix!
